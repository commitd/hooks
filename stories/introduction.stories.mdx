import { Meta } from '@storybook/addon-docs/blocks'
import { version } from '../package.json'

<Meta title="Introduction" />

# Committed Hooks

<code>v{version}</code>

[![Committed Badge](https://img.shields.io/endpoint?url=https%3A%2F%2Fcommitted.software%2Fbadge)](https://committed.io)
![Build Status](https://github.com/commitd/hooks/workflows/build/badge.svg?branch=main)
[![Quality Gate Status](https://sonarcloud.io/api/project_badges/measure?project=commitd_hooks&metric=alert_status)](https://sonarcloud.io/dashboard?id=commitd_hooks)
[![Coverage](https://sonarcloud.io/api/project_badges/measure?project=commitd_hooks&metric=coverage)](https://sonarcloud.io/dashboard?id=commitd_hooks)
![GitHub repo size](https://img.shields.io/github/repo-size/commitd/hooks)

This is a library of useful react hooks.

It aims to remove repetition and effort for our different react based projects.
They will be updated as we continue to improve and evolve our use of react.

## Getting Started

```
yarn add @committed/hooks
```

Each hook can be imported from the package directly or the matching names folder e.g..

```typescript
import { useEventListener } form '@committed/hooks'
```

or

```typescript
import { useEventListener } form '@committed/hooks/useEventListener'
```

The later may help with tree-shaking depending on your setup.

## Links

Here are a few helpful links for getting started with hooks:

- [Tutorial](https://reactjs.org/docs/hooks-overview.html) Learn about react hooks.
- [Hooks Thinking](https://wattenberger.com/blog/react-hooks) Learn how to think about hooks.
- [Code](https://github.com/commitd/hooks) See the source code on GitHub.
