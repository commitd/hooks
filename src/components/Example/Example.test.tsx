import React from 'react'
import { render } from '../../setupTests'
import { Primary } from './Example.stories'

it('renders story', () => {
  const { asFragment } = render(<Primary />)
  expect(asFragment()).toMatchSnapshot()
})
