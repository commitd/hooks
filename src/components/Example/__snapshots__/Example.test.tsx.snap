// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders story 1`] = `
<DocumentFragment>
  <div>
    Example
  </div>
</DocumentFragment>
`;
